packages/cosmic-swingset/lib/ag-solo/vats/lib-dehydrate.js

```diff
@@ -0,0 +1,82 @@
+import harden from '@agoric/harden';
+import { makeMarshal } from '@agoric/marshal';
+import makeStore from '@agoric/store';
+import { assert, details } from '@agoric/assert';
+
+// Marshalling for the UI should only use the user's petnames. We will
+// call marshalling for the UI "dehydration" and "hydration" to distinguish it from
+// other marshalling.
+export const makeDehydrator = (initialUnnamedCount = 0) => {
+  let unnamedCount = initialUnnamedCount;
+
+  const petnameKindToMappings = makeStore();
+
+  const searchOrder = [];
+
+  const makeMappings = kind => {
+    assert.typeof(kind, 'string', details`kind ${kind} must be a string`);
```

Comment:
If you omit the details here, `assert.typeof` defaults to almost exactly the same thing.

Reply:
I think it does not print `kind`, though, which I'd like to keep.

Reply:
ok